dask/dataframe/core.py

```diff
@@ -183,6 +183,13 @@ def __getstate__(self):
     def __setstate__(self, state):
         self.dask, self._name, self._meta = state
 
+    def __bool__(self):
+        raise TypeError(
+            "Dask dataframe Scalars don't support conversion to a "
+            "boolean because they are lazily evaluated. Consider computing "
+            "the scalar value prior to converting to a boolean."
+        )
```

Comment:
Often this happens when users write something like the following:

```python
if df.x.all():
    ...
```

Because the boolean conversion is implicit, less experienced users may not understand that they have indeed tried to convert the object to a boolean value. Do you have any thoughts on how this error message might better address novice users in this case?

> Because Dask objects are lazily we can not implicitly evaluate them in boolean conditions like if statements.  Try calling `.compute()` to force computation before converting this to a boolean value.

?  

We can probably do better than this, but I thought I'd put something out.

Reply:
Thanks for the feedback, I updated to error message to reference conditional statements and also include the `repr` for the index that's being converted to a boolean. I think it will be more clear to novice users than what I had before. Let me know what you think, I'm happy to keep iterating

Reply:
Looks great to me.  Thanks for the extra attention here @jrbourbeau !